haystack/core/component/component.py

```diff
@@ -157,6 +157,13 @@ def __call__(cls, *args, **kwargs):
         # We use this flag to check that.
         instance.__haystack_added_to_pipeline__ = None
 
+        # Only Components with variadic inputs can be greedy. If the user set the greedy flag
+        # to True, but the component doesn't have a variadic input, we set it to False.
+        # We can have this information only at instance creation time, so we do it here.
+        is_variadic = any(socket.is_variadic for socket in instance.__haystack_input__._sockets_dict.values())
+        if not is_variadic and cls.__haystack_is_greedy__:
+            cls.__haystack_is_greedy__ = False
```

Comment:
Not sure I'd do this, I would rather log a warning. If you recall, this "let them fail" philosophy was already used in 2.x in other parts of the code.